### PR TITLE
Add clang-format support

### PR DESCRIPTION
**Pull request by andronat**
Added support for clang-format across the project with Mode: LLVM.

### CODE REVIEW

**Comment by ccadar** [comment]
@andronat Thanks.  But I don't really like this patch, I think it unnecessarily pollutes the version history. 
This being said, I'm happy to use `clang-format`, but I'd rather incrementally reformat code as we go.  What do others think?

**Comment by parras** [comment]
> @andronat Thanks. But I don't really like this patch, I think it unnecessarily pollutes the version history.
> This being said, I'm happy to use `clang-format`, but I'd rather incrementally reformat code as we go. What do others think?

I'm also in favour of incrementally reformatting.

**Comment by andronat** [comment]
I was concerned that the repo history will be full of commits like "Format file X" or even worse commits with significant changes and formatting together. So I thought that if I make 1 commit with everything together we can avoid all the above.

Are you sure you want me to revert the above?

**Comment by andronat** [comment]
For example in PR #50 I'm changing 3 files. That means the next PR will need to have commits says "Format file X" and so on.

**Comment by ccadar** [comment]
@andronat When I say incrementally I mean only the patch code (and surrounding code when needed).  Not entire files.  Note that you can configure by default to have `clang-format` format only your patch.

**Comment by daniel-grumberg** [comment]
Look into git-clang format. That way code only get's reformatted when it is changed. This is what I use.

**Comment by andronat** [comment]
Removed the reformatting. This is ready to go in.

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 # Anything pertaining to building the project
 build/
+.vscode/
```
Comment (daniel-grumberg) [inline .gitignore:6]: I don't like this here this should be in your personal gitignore. The way you do this is `git config --global core.excludesfile /path/to/your/personal/gitignore`

Comment (andronat) [reply .gitignore:6]: Isn’t your suggestion excluding the folder from every project i’m working with? Some projects have ‘.vscode’ in their git. 

Additionally, isn’t this instantly give a better experience to anyone that downloads the repo and they don’t have to battle with their gitignore?

**Comment by ccadar** [comment]
@daniel-grumberg Oops, I somehow didn't see your objection to `.vscode`.  Feel free to revert and/or submit another patch if needed.

**Comment by daniel-grumberg** [comment]
It's fine I don't care that strongly

**Pull request merged**